app/src/full/java/io/homeassistant/companion/android/settings/wear/views/SettingsWearHomeView.kt

```diff
@@ -48,30 +50,42 @@ fun LoadSettingsHomeView(
                     hasData = hasData,
                     isAuthed = isAuthenticated,
                     navigateFavorites = { navController.navigate(SettingsWearMainView.FAVORITES) },
-                    navigateTemplateTile = { navController.navigate(SettingsWearMainView.TEMPLATE) },
+                    navigateTemplateTile = { navController.navigate(SettingsWearMainView.TEMPLATES) },
                     loginWearOs = loginWearOs,
                     onBackClicked = onStartBackClicked,
                     events = settingsWearViewModel.resultSnackbar
                 )
             }
-            composable(SettingsWearMainView.TEMPLATE) {
-                SettingsWearTemplateTile(
-                    template = settingsWearViewModel.templateTileContent.value,
-                    renderedTemplate = settingsWearViewModel.templateTileContentRendered.value,
-                    refreshInterval = settingsWearViewModel.templateTileRefreshInterval.value,
-                    onContentChanged = {
-                        settingsWearViewModel.setTemplateContent(it)
-                        settingsWearViewModel.sendTemplateTileInfo()
-                    },
-                    onRefreshIntervalChanged = {
-                        settingsWearViewModel.templateTileRefreshInterval.value = it
-                        settingsWearViewModel.sendTemplateTileInfo()
+            composable(SettingsWearMainView.TEMPLATES) {
+                SettingsWearTemplateTileList(
+                    templateTiles = settingsWearViewModel.templateTiles,
+                    onTemplateTileClicked = { tileId ->
+                        navController.navigate(SettingsWearMainView.TEMPLATE_TILE.format(tileId))
                     },
                     onBackClicked = {
                         navController.navigateUp()
                     }
                 )
             }
+            composable(
+                SettingsWearMainView.TEMPLATE_TILE.format("{tileId}"),
+                arguments = listOf(navArgument("tileId") { type = NavType.IntType })
+            ) { backStackEntry ->
+                val tileId = backStackEntry.arguments?.getInt("tileId")
+                val templateTile = settingsWearViewModel.templateTiles[tileId]
+
+                // TODO: error handling
+                templateTile?.let {
+                    SettingsWearTemplateTile(
+                        template = it.first,
+                        // TODO: use rendered template
+                        renderedTemplate = "",
+                        refreshInterval = it.second,
+                        onContentChanged = {},
+                        onRefreshIntervalChanged = {}
+                    ) {}
+                }
+            }
```

Comment:
Thoughts about this part?

My idea was to change the current "Template tile" row to "Templates tiles". When clicked, it opens a list of all current Template tiles and a button for adding a new one. When a Template tile is clicked on in the list, the current `SettingsWearTemplateTile` is opened.

Reply:
how would it work to add a template tile here? Dont we need a tile ID to apply the template to it?

Reply:
> how would it work to add a template tile here? Dont we need a tile ID to apply the template to it?

I had misunderstood how things work. The way it works is by the user adding a new Tile on the Wear OS device (or in the Wear device settings app on the phone), and only then it will become visible in the Home Assistant phone/wear app settings. So no button for adding a new Template tile needed here.

Reply:
And big thanks to @marazmarci for figuring most of this out for us :grin: :+1: